### PR TITLE
docs: fix Adam paper link and correct grammatical error in fused_adam.py

### PR DESCRIPTION
**Pull request by ishanjmukherjee**
Fixes a broken paper link in `fused_adam.py` and cleans up a grammar typo:

- The reST target label now matches the inline text (`Adam:` instead of `Adam -`), restoring the link. 
- "Adam was been proposed [...]” -> “Adam was proposed [...]”

Docs-only change, no code or behavior changes.

### CODE REVIEW

**Review by loadams — APPROVED** [review]

File: deepspeed/ops/adam/fused_adam.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ class FusedAdam(torch.optim.Optimizer):
         set_grad_none (bool, optional): whether set grad to None when zero_grad()
             method is called. (default: True)
 
-    .. _Adam - A Method for Stochastic Optimization:
+    .. _Adam: A Method for Stochastic Optimization:
```
Comment (loadams) [inline deepspeed/ops/adam/fused_adam.py:70]: Will this work properly with the docs given the `:` at the end of the title of this paper?

Comment (loadams) [reply deepspeed/ops/adam/fused_adam.py:70]: @ishanjmukherjee - another tag on this, thoughts on the impact of the colon here?

**Comment by loadams** [comment]
@ishanjmukherjee - closing as stale due to no response on the PR feedback.  If you see this, let us know and we can re-activate and get this merged.  Thanks!
